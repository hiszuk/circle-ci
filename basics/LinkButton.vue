<template>
  <a :class="`button--${type}`" @click="clickLink">
    {{ label }}
  </a>
</template>

<script>
export default {
  name: 'LinkButton',
  props: {
    type: {
      type: String,
      default: 'grey',
      required: false
    },
    label: {
      type: String,
      default: '',
      required: true
    }
  },
  methods: {
    clickLink() {
      this.$emit('click')
    }
  }
}
</script>