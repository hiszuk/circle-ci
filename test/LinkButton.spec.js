import { shallowMount } from '@vue/test-utils'
import LinkButton from '@/basics/LinkButton.vue'

describe('LinkButton.vue', () => {
  const propsData = {
    type: 'blue',
    label: 'ButtonName'
  }

  describe('props', () => {
    test('propsの型が合っているか(label)', () => {
      const wrapper = shallowMount(LinkButton, { propsData })
      expect(typeof wrapper.vm.$props.label).toBe('string')
    })
  })

  describe('props', () => {
    test('propsの型が合っているか(type)', () => {
      const wrapper = shallowMount(LinkButton, { propsData })
      expect(typeof wrapper.vm.$props.type).toBe('string')
    })
  })

  describe('props', () => {
    test('propsが必須となっているか(label)', () => {
      const wrapper = shallowMount(LinkButton, { propsData })
      expect(wrapper.vm.$options.props.label.required).toBe(true)
    })
  })

  describe('props', () => {
    test('propsが必須となっていない(type)', () => {
      const wrapper = shallowMount(LinkButton, { propsData })
      expect(wrapper.vm.$options.props.type.required).toBe(false)
    })
  })

  describe('props', () => {
    test('propsを受け取れること(label)', () => {
      const wrapper = shallowMount(LinkButton, { propsData })
      expect(wrapper.vm.$props.label).toBe(propsData.label)
    })
  })

  describe('props', () => {
    test('propsを受け取れること(type)', () => {
      const wrapper = shallowMount(LinkButton, { propsData })
      expect(wrapper.vm.$props.type).toBe(propsData.type)
    })
  })

  describe('template', () => {
    it('snapshot', () => {
      const wrapper = shallowMount(LinkButton, { propsData })
      // レンダリング結果がスナップショットと同じか
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

})
