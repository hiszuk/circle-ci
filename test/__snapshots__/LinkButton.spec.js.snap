// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton.vue template snapshot 1`] = `
<a
  class="button--blue"
>
  
  ButtonName

</a>
`;
